sdk/iotcentral/Microsoft.Azure.Management.IotCentral/src/Generated/AppsOperations.cs

```diff
@@ -292,10 +287,10 @@ public async Task<AzureOperationResponse<App>> CreateOrUpdateWithHttpMessagesAsy
         /// <param name='cancellationToken'>
         /// The cancellation token.
         /// </param>
-        public async Task<AzureOperationResponse<App>> UpdateWithHttpMessagesAsync(string resourceGroupName, string resourceName, AppPatch appPatch, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationHeaderResponse<AppsUpdateHeaders>> UpdateWithHttpMessagesAsync(string resourceGroupName, string resourceName, AppPatch appPatch, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
         {
             // Send Request
-            AzureOperationResponse<App> _response = await BeginUpdateWithHttpMessagesAsync(resourceGroupName, resourceName, appPatch, customHeaders, cancellationToken).ConfigureAwait(false);
+            AzureOperationHeaderResponse<AppsUpdateHeaders> _response = await BeginUpdateWithHttpMessagesAsync(resourceGroupName, resourceName, appPatch, customHeaders, cancellationToken).ConfigureAwait(false);
             return await Client.GetPutOrPatchOperationResultAsync(_response, customHeaders, cancellationToken).ConfigureAwait(false);
```

Comment:
@Yao725, autorest appears to be generating this incorrectly. `GetPutOrPatchOperationResultAsync` returns an `AzureOperationResponse` but the method in this `AppsOperations` class expects an `AzyreOperationHeaderResponse`. Do you know who to contact about this?

Reply:
@Yao725, for this one I think the problem is that the AzureClientExtensions class is missing a method definition for:

  ```
  public static async Task<AzureOperationHeaderResponse<THeader>> GetPutOrPatchOperationResultAsync<THeader>(this IAzureClient client, AzureOperationHeaderResponse<THeader> response, Dictionary<string, List<string>> customHeaders, CancellationToken cancellationToken)
  ```

Reply:
I think this problem is also from the swagger file. According to the ARM RPC, a patch method should have a response of [200](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#response-1), so, is there any reason that this patch method doesn't have this [response](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/iotcentral/resource-manager/Microsoft.IoTCentral/preview/2021-11-01-preview/iotcentral.json#L192-L217)? I tried to manually add a 200 response and the generated code is what we are expecting:
```csharp
AzureOperationResponse<App,AppsUpdateHeaders> _response = await BeginUpdateWithHttpMessagesAsync(resourceGroupName, resourceName, appPatch, customHeaders, cancellationToken).ConfigureAwait(false);
```

Reply:
This patch operation is asynchronous so 202 is returned.

Reply:
Yeah, I understand that it needs 202 because this operation is asynchronous. But I think it should both follow the docs that you and I mentioned. As 200 is missed, what will happen if you call the patch method but nothing is actually updated? I think return 200 is more reasonable in this scenario.